backend/src/test/java/com/example/parkingspot/controller/ZoneControllerMvcTest.java

```diff
@@ -0,0 +1,39 @@
+package com.example.parkingspot.controller;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.RequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import com.example.parkingspot.service.EventService;
+import com.example.parkingspot.service.ZoneService;
+
+@WebMvcTest(ZoneController.class)
+@AutoConfigureMockMvc(addFilters = false)
+public class ZoneControllerMvcTest {
+
+  @MockBean
+  private ZoneService zoneService;
+
+  @MockBean
+  private EventService eventService;
+
+  @Autowired
+  private MockMvc mockMvc;
+
```

Comment:
Missing annotation @Test?

Reply:
Cannot invoke "org.springframework.test.web.servlet.ResultMatcher.match(org.springframework.test.web.servlet.MvcResult)" 
because "matcher" is null

When runnning this test. Does it pass for you?

Reply:
This cuz i was to "fast" and lazy, missed to just push unit tests for ZoneController. Started with MVC tests before i realized and removed the @Test annotation you are pointing out is missing. 

So this PR is only for unit tests of Java code and not MVC

Reply:
Ah, ok